### ISSUE TITLE
ping plugin: rrd file incorrect

### ISSUE THREAD

**Issue by joelfranco** [collectd/collectd#2720]
*   Version of collectd: 5.4.1-6+deb8u1
*   Operating system / distribution: debian 8.7 (jessie)

I use the collection.cgi to view the rrd graphs. I've configurated the ping plugin in collectd as:

LoadPlugin ping
<Plugin ping>
        Host "www.uol.com.br"
        Host "8.8.8.8"
        Interval 1.0
        Timeout 0.9
        TTL 255
        Device "ppp1"
        #MaxMissed -1
</Plugin>

The graphs are created in the correct dir: 

$ ls -l /var/lib/collectd/rrd/sol1.gory.local/ping/
total 888
-rw-r--r-- 1 root root 148648 Mar 22 11:14 ping-8.8.8.8.rrd
-rw-r--r-- 1 root root 148648 Mar 22 11:14 ping_droprate-8.8.8.8.rrd
-rw-r--r-- 1 root root 148648 Mar 22 11:14 ping_droprate-www.uol.com.br.rrd
-rw-r--r-- 1 root root 148648 Mar 22 11:14 ping_stddev-8.8.8.8.rrd
-rw-r--r-- 1 root root 148648 Mar 22 11:14 ping_stddev-www.uol.com.br.rrd
-rw-r--r-- 1 root root 148648 Mar 22 11:14 ping-www.uol.com.br.rrd

But when i try to view via web using collection.cgi the ping graph is broken.

Debugging the collection.cgi:

$ /usr/lib/cgi-bin/collection.cgi 'http://192.168.215.1/cgi-bin/collection.cgi?action=show_graph;plugin=ping;type=ping;timespan=hour;type_instance=8.8.8.8;host=sol1.gory.local'
Content-Type: image/png; charset=ISO-8859-1

<h1>Software error:</h1>
<pre>RRDs::graph: No DS called 'ping' in '/var/lib/collectd/rrd/sol1.gory.local/ping/ping-8.8.8.8.rrd' at /usr/lib/cgi-bin/collection.cgi line 787.
</pre>
<p>
For help, please send mail to this site's webmaster, giving this error message 
and the time and date of the error.

</p>
[Thu Mar 22 11:16:11 2018] collection.cgi: RRDs::graph: No DS called 'ping' in '/var/lib/collectd/rrd/sol1.gory.local/ping/ping-8.8.8.8.rrd' at /usr/lib/cgi-bin/collection.cgi line 787.

It says: No DS called 'ping' ..

Using the rrdinfo tool:

$ rrdinfo /var/lib/collectd/rrd/sol1.gory.local/ping/ping-8.8.8.8.rrd  | grep ds
ds[value].index = 0
ds[value].type = "GAUGE"
ds[value].minimal_heartbeat = 20
ds[value].min = 0.0000000000e+00
ds[value].max = 6.5535000000e+04
ds[value].last_ds = "24.741200"
ds[value].value = 9.8964800000e+01
ds[value].unknown_sec = 0

The ds created is 'value' when it is supposed to be 'ping'?

Maybe it's the cause of the collection.cgi do not find the graph pings.

It's really a bug?

If yes, the bug is in the graph creation (ping.c)?

Thank you in advance.

Joel

**Comment by rpv-tomsk** [comment]
>The ds created is 'value' when it is supposed to be 'ping'?

No, it has to be `value`.  Please look into https://github.com/collectd/collectd/commit/147690b3baf04341851757c9424596ceb37c45ba and https://github.com/collectd/collectd/commit/6c1415d484e5913ac5a7341528378e9a98ee515d

There has to be following configuration block in your `collection.conf`:
```
<Type ping>
  DataSources value
  DSName "value Latency"
  RRDTitle "Network latency ({type_instance})"
  RRDVerticalLabel "Milliseconds"
  RRDFormat "%5.2lfms"
</Type>
```

You have to correct the `DataSources` option value.

**Comment by joelfranco** [comment]
Please, observe that I'm using collection.cgi, and not the collection3 system, that is more complex.

To run, i've fixed changing:

oryjoel@peru:~$ sudo diff /usr/share/doc/collectd/examples/collection.cgi /usr/lib/cgi-bin/collection.cgi 
2081,2083c2081,2103
<     ping => ['DEF:ping_avg={file}:ping:AVERAGE',
<     'DEF:ping_min={file}:ping:MIN',
<     'DEF:ping_max={file}:ping:MAX',
---
>     ping => ['DEF:ping_avg={file}:value:AVERAGE',
>     'DEF:ping_min={file}:value:MIN',
>     'DEF:ping_max={file}:value:MAX',
>     "AREA:ping_max#$HalfBlue",
>     "AREA:ping_min#$Canvas",
>     "LINE1:ping_avg#$FullBlue:Ping",
>     'GPRINT:ping_min:MIN:%4.1lf ms Min,',
>     'GPRINT:ping_avg:AVERAGE:%4.1lf ms Avg,',
>     'GPRINT:ping_max:MAX:%4.1lf ms Max,',
>     'GPRINT:ping_avg:LAST:%4.1lf ms Last'],
>     ping_droprate => ['DEF:ping_avg={file}:value:AVERAGE',
>     'DEF:ping_min={file}:value:MIN',
>     'DEF:ping_max={file}:value:MAX',
>     "AREA:ping_max#$HalfBlue",
>     "AREA:ping_min#$Canvas",
>     "LINE1:ping_avg#$FullBlue:Ping",
>     'GPRINT:ping_min:MIN:%4.1lf ms Min,',
>     'GPRINT:ping_avg:AVERAGE:%4.1lf ms Avg,',
>     'GPRINT:ping_max:MAX:%4.1lf ms Max,',
>     'GPRINT:ping_avg:LAST:%4.1lf ms Last'],
>     ping_stddev => ['DEF:ping_avg={file}:value:AVERAGE',
>     'DEF:ping_min={file}:value:MIN',
>     'DEF:ping_max={file}:value:MAX',
oryjoel@peru:~$
